File: test_commits/python/feature_addition_changed5.py
from datetime import datetime

class Task:
    def __init__(self, name, estimated_hours=0, deadline=None):
        self.name = name
        self.completed = False
        self.estimated_hours = estimated_hours
        self.deadline = datetime.strptime(deadline, "%Y-%m-%d") if deadline else None
        self.completed_at = None

    def mark_complete(self):
        self.completed = True
        self.completed_at = datetime.now()

    def is_overdue(self):
        return self.deadline and not self.completed and datetime.now() > self.deadline

class Project:
    def __init__(self, title):
        self.title = title
        self.tasks = []

    def add_task(self, name, estimated_hours=0, deadline=None):
        self.tasks.append(Task(name, estimated_hours, deadline))

    def get_completed_tasks(self):
        return [task for task in self.tasks if task.completed]

    def get_progress(self):
        total = len(self.tasks)
        completed = len(self.get_completed_tasks())
        return round((completed / total) * 100) if total > 0 else 0

class ProjectManager:
    def __init__(self):
        self.projects = []

    def add_project(self, title):
        self.projects.append(Project(title))

    def get_all_projects(self):
        return self.projects

    def get_overall_progress(self):
        if not self.projects:
            return 0
        return round(sum(p.get_progress() for p in self.projects) / len(self.projects))

File: test_commits/javascript/feature_addition_changed5.js
class Task {
    constructor(name, estimatedHours = 0, deadline = null) {
        this.name = name;
        this.completed = false;
        this.estimatedHours = estimatedHours;
        this.deadline = deadline ? new Date(deadline) : null;
        this.completedAt = null;
    }

    markComplete() {
        this.completed = true;
        this.completedAt = new Date();
    }

    isOverdue() {
        if (!this.completed && this.deadline) {
            return new Date() > this.deadline;
        }
        return false;
    }
}

class Project {
    constructor(title) {
        this.title = title;
        this.tasks = [];
    }

    addTask(name, estimatedHours = 0, deadline = null) {
        this.tasks.push(new Task(name, estimatedHours, deadline));
    }

    getCompletedTasks() {
        return this.tasks.filter(task => task.completed);
    }

    getProgress() {
        const total = this.tasks.length;
        const completed = this.getCompletedTasks().length;
        return total === 0 ? 0 : Math.round((completed / total) * 100);
    }
}

class ProjectManager {
    constructor() {
        this.projects = [];
    }

    addProject(title) {
        this.projects.push(new Project(title));
    }

    getAllProjects() {
        return this.projects;
    }

    getOverallProgress() {
        const projectProgress = this.projects.map(p => p.getProgress());
        if (projectProgress.length === 0) return 0;
        return Math.round(projectProgress.reduce((a, b) => a + b, 0) / projectProgress.length);
    }
}
